frontend/lib/src/components/elements/Spinner/Spinner.tsx

```diff
@@ -31,18 +31,26 @@ export interface SpinnerProps {
 }
 
 function Spinner({ width, element }: SpinnerProps): ReactElement {
-  const theme: EmotionTheme = useTheme()
   const { activeTheme } = React.useContext(LibContext)
   const usingCustomTheme = !isPresetTheme(activeTheme)
-  const styleProp = { width }
+  const { cache } = element
+  const styleProp = cache
+    ? {
+        width,
+        background:
+          "linear-gradient(to bottom, #FFFFFF 0%, #FFFFFF 80%, transparent 100%)",
+        padding: "0px 0px 16px 0px",
+      }
+    : { width }
 
   return (
-    <div className="stSpinner" data-testid="stSpinner" style={styleProp}>
+    <div
+      className={`stSpinner ${cache ? "cacheSpinner" : ""}`}
```

Comment:
Putting these together with conditional string formatting didn't seem ideal to me, but after searching the rest of our code, the alternatives (using a `classNames` function) aren't that much better, though it is maybe worth switching for consistency.